fuseimg/data/ops/shape_ops.py

```diff
@@ -82,7 +81,214 @@ def __call__(self, sample_dict: NDict, op_id: Optional[str], key: str,
+                                                                 int(inner_image_width * resize_ratio)),
+                                                   mode='reflect',
+                                                   anti_aliasing=True
+                                                   )
+        return inner_image
+    
+    def __call__(self, sample_dict: NDict, key: str , resize_to: Tuple,
+                 padding: Tuple
+        ):
+        '''
+        :param resize_to:               new size of input images, keeping proportions
+        :param padding:                 padding size
+        ''' 
+        
+        img = sample_dict[key]
+        # resize
+        if resize_to is not None:
+            img = self.resize_image(img, resize_to)
+
+        # padding
+        if padding is not None:
+            img = self.pad_image(img, padding, resize_to)
+        sample_dict[key] = img
+        return sample_dict
+        
+class OpRemoveDarkBackgroundRectangle2D(OpBase):
+    '''
+     removes dark background in a rectangle patch from a 2d image
+    '''
+    def __init__(self, dark_area_threshold : int = 10, blocks_num : int = 30, **kwargs):
+        super().__init__(**kwargs)
+        self.dark_area_threshold = dark_area_threshold
+        self.blocks_num = blocks_num
+
+    def find_breast_aabb(self,img : np.ndarray) -> Tuple[int ,int, int, int]:
+        """
+        split the images into blocks, each block containing (1/30 x 1/30) of the image.
+        All blocks above a threshold (10) are considered non-empty.
+        Then, the biggest connected component (at the blocks level) is extracted, and its axis-aligned bbox is returned.
+        :param img: Image instance , expected 2d breast integer grayscale image where 0 is black background color
+        :param dark_area_threshold: defines grayscale level from which lower is consider dark / outside of body
+        :param blocks_num: number of blocks the algorithm split the images into
+        :return: four coordinates
+        """
+        bl_rows = img.shape[0]//self.blocks_num
+        bl_cols = img.shape[1]//self.blocks_num
+
+        rows_starts = list(range(self.blocks_num))
+        cols_starts = list(range(self.blocks_num))
+
+        rows_starts = list(map(lambda x:x*bl_rows, rows_starts))
+        cols_starts = list(map(lambda x: x * bl_cols, cols_starts))
+
+        cells = np.zeros((self.blocks_num,self.blocks_num))
+
+        for ri,r in enumerate(rows_starts):
+            for ci,c in enumerate(cols_starts):
+                r_end = min(img.shape[0]-1,r+bl_rows)
+                c_end = min(img.shape[1] - 1, c + bl_cols)
+                cells[ri,ci] = np.mean(img[r:r_end,c:c_end])
+
+        cells_binary = np.zeros((self.blocks_num,self.blocks_num),dtype=bool)
+        cells_binary[cells>self.dark_area_threshold] = True #TODO: this is a hardcoded threshold, see if there's a better alternative
```

Comment:
missed TODO here

Reply:
it's something we inherit from old code, you want to just remove it ?